### PR TITLE
[EDNA-55] Define Dashboard API

### PR DESCRIPTION
**Pull request by gromakovsky**
## Description

Problem: we are going to implement Dashboard page, but currently there
is no backend API for it.

Solution: define API types and schema. Endpoints will be implemented
later, currently we use stubs with 501 HTTP code.

Note: currently we don't define any endpoints necessary to drop points
and create new sub-experiments. They will be added later.

image

## Related issue(s)

https://issues.serokell.io/issue/EDNA-55

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

We don't have API tests for now.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: backend/src/Edna/Dashboard/Web/API.hs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      :> Get '[JSON] ExperimentsResp
+
+  , -- | Get sub-experiment's (meta-)data by ID.
+    deGetSubExperiment :: route
+      :- "subExperiment"
+      :> Summary "Get sub-experiment's (meta-)data by ID"
+      :> Capture "subExperimentId" SubExperimentId
+      :> Get '[JSON] (WithId 'SubExperimentId SubExperimentResp)
+
+  , -- | Get sub-experiment's (meta-)data by ID.
```
Comment (pva701) [inline backend/src/Edna/Dashboard/Web/API.hs]: wrong comment

File: backend/src/Edna/Dashboard/Web/Types.hs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import Edna.Util
+  (CompoundId, IdType(..), MethodologyId, ProjectId, SubExperimentId, TargetId, ednaAesonWebOptions,
+  gDeclareNamedSchema)
+import Edna.Web.Types (WithId)
+
+-- | Experiment as response from the server.
+data ExperimentsResp = ExperimentsResp
+  { erExperiments :: [WithId 'ExperimentId ExperimentResp]
+  , erMeanIC50 :: Maybe Double
+  -- ^ If compound and target and selected,
```
Comment (pva701) [inline backend/src/Edna/Dashboard/Web/Types.hs]: are selected maybe?

**Review by pva701 — APPROVED** [review]
Looks more or less sane, but it's hard to predict precisely what else will be needed. AFAIU, there are no creation methods (like to create a new subexperiment?).

**Comment by gromakovsky** [comment]
@pva701 

> AFAIU, there are no creation methods (like to create a new subexperiment?).

Yes, that's written in the PR description:
> Note: currently we don't define any endpoints necessary to drop points and create new sub-experiments. They will be added later.

The reason is that I just follow our roadmap (that I invented myself), according to which point dropping will be implemented after we are done with Dashboard that doesn't have it.

**Pull request merged**